core/rfb.js

```diff
@@ -1603,6 +1616,33 @@ export default class RFB extends EventTargetMixin {
         RFB.messages.xvpOp(this._sock, ver, op);
     }
 
+    _updateCursor(rgba, x, y, w, h) {
+        this._cursorImage = {
+            rgbaPixels: rgba,
+            x: x, y: y, w: w, h: h,
+        }
+        this._refreshCursor();
+    }
+
+    _shouldShowDot() {
+        if (!this._showDot) { return false; }
+        for (let y = 0; y < this._cursorImage.h; y++) {
+            for (let x = 0; x < this._cursorImage.w; x++) {
+                let idx = 4 * (x + this._cursorImage.w * y) + 3;
```

Comment:
why not just do 

```js
for (let i = 3; i < this._cursorImage.length; i += 4) {
   if (this_cursorImage.rgbaPixels[idx]) { return false }
}
```

Reply:
also, this needs comments as to why it's doing what it's doing

Reply:
I agree